### PR TITLE
Move the library to actual enums instead of constants.

### PR DESCRIPTION
**Pull request by RyadaProductions**
PR for #14 

Changes:
- Upgraded lowest .NET Standard version to 1.3
- Added System.Reflection.TypeExtensions and System.ComponentModel.Primitives for .NET Standard 1.3
- Changed all constants/static strings to enums
- Added extension method to get the enum description value on enums.

Suggestions for further improvements:
- Limit the string parameters in EndpointMethods to the actual enums for a safer API

### CODE REVIEW

**Comment by RyadaProductions** [comment]
I changed the PR as a showcase of the current version I managed to make with the changes lied out in #14 
This is all open to change as we have some caveats we still have to discuss. Especially regarding for example `Maps.cs.dt` which will have enum name collisions if we only use the mapname as the enum identifier. Currently i worked around it by doing `mapname + "_" + notes` but this doesn't seem to be completely feasible although workable. 

Also should we keep the `SNAKE_CASE` or should we try to convert to the C# standard of using `PascalCase` in enums?

**Comment by MingweiSamuel** [comment]
PascalCase is fine when it can be used, but for the `ToString` I guess it should be SNAKE_CASE so we dont need to reformat

**Comment by RyadaProductions** [comment]
I would say lets keep it consistent and in `SNAKE_CASE` then. Even if it isn't the standard.

File: Camille/gen/download.ps1 [hunk 1]
```diff
@@ -6,3 +6,11 @@ $DD = ((Invoke-WebRequest -Uri "http://ddragon.leagueoflegends.com/realms/na.jso
...
 
 Invoke-WebRequest -Uri "http://ddragon.leagueoflegends.com/cdn/$DD/data/en_US/champion.json" -OutFile .\.champion.json
+
+echo "Downloading static data"
+
+Invoke-WebRequest -Uri "http://static.developer.riotgames.com/docs/lol/seasons.json" -OutFile .\Enums\.seasons.json
+Invoke-WebRequest -Uri "http://static.developer.riotgames.com/docs/lol/queues.json" -OutFile .\Enums\.queues.json
+Invoke-WebRequest -Uri "http://static.developer.riotgames.com/docs/lol/maps.json" -OutFile .\Enums\.maps.json
+Invoke-WebRequest -Uri "http://static.developer.riotgames.com/docs/lol/gameModes.json" -OutFile .\Enums\.gameModes.json
+Invoke-WebRequest -Uri "http://static.developer.riotgames.com/docs/lol/gameTypes.json" -OutFile .\Enums\.gameTypes.json
```
Comment (MingweiSamuel) [inline Camille/gen/download.ps1]: Actually `download.ps1` isn't used anymore, I moved the downloads to `Camille.csproj` to be more os-agnostic and seems I forgot to remove this

Comment (RyadaProductions) [reply Camille/gen/download.ps1]: I removed `downlaod.ps1` and moved the added download endpoints to `Camille.csproj`

**Comment by MingweiSamuel** [comment]
Is this still [WIP]?

**Comment by RyadaProductions** [comment]
Not anymore i forgot to update the PR to reflect the state. It should be done now.

**Pull request merged**